src/policy/v3_policy.cpp

```diff
@@ -0,0 +1,194 @@
+
+    // Now we have all ancestors, so we can start checking v3 rules.
+    if (ptx->nVersion == 3) {
+        // v3 transactions can have at most 1 unconfirmed parent
+        if (mempool_ancestors.size() + in_package_parents.size() > 1) {
+            return strprintf("tx %s would have too many ancestors", ptx->GetWitnessHash().ToString());
+        }
+
+        const bool has_parent{mempool_ancestors.size() + in_package_parents.size() > 0};
+        if (has_parent) {
+            // A v3 child cannot be too large.
+            if (vsize > V3_CHILD_MAX_VSIZE) {
+                return strprintf("v3 child tx %s is too big: %u > %u virtual bytes",
+                                 ptx->GetWitnessHash().ToString(), vsize, V3_CHILD_MAX_VSIZE);
+            }
+
+            // Find the parent and extract the information we need for v3 checks.
+            int parent_version = 0;
+            // fixme: parent hash is txid, should return err strings with wtxid?
+            Txid parent_hash = Txid::FromUint256(uint256(0));
+            Wtxid parent_wtxid = Wtxid::FromUint256(uint256(0));
+            int other_mempool_descendants = 0;
+
+            if (mempool_ancestors.size() > 0) {
+                // There's a parent in the mempool.
+                auto& parent = *mempool_ancestors.begin();
+                parent_version = parent->GetTx().nVersion;
+                // Subtract 1 because descendant count is inclusive of the tx itself.
+                other_mempool_descendants = parent->GetCountWithDescendants() - 1;
+                parent_hash = parent->GetTx().GetHash();
+                parent_wtxid = parent->GetTx().GetWitnessHash();
+            } else {
+                // Ancestor must be in the package. Find it.
+                auto &parent_index = in_package_parents[0];
+                // If the in-package parent has mempool ancestors, then this is
+                // a v3 violation.
+                if (package_with_ancestors.ancestor_counts[parent_index] > 0) {
+                    return strprintf("tx %s would have too many ancestors", ptx->GetWitnessHash().ToString());
+                }
+
+                auto &parent = package_with_ancestors.package[parent_index];
+                parent_version = parent->nVersion;
+                // There may be in-package descendants, which we will look for below.
+                other_mempool_descendants = 0;
+                parent_hash = parent->GetHash();
+                parent_wtxid = parent->GetWitnessHash();
+            }
+
+            // If there is a parent, it must have the right version.
+            if (parent_version != 3) {
+                return strprintf("v3 tx %s cannot spend from non-v3 tx %s",
+                                 ptx->GetWitnessHash().ToString(), parent_wtxid.ToString());
+            }
+
+            // The mempool or in-package parent cannot have any other in-mempool children.
```

Comment:
Was thinking about why this is ok. AcceptMultipleTransactions doesn't allow RBF, and future package RBF should work with this?

```suggestion
            // The mempool or in-package parent cannot have any other in-mempool children.
            // We don't need to handle the "single RBF" case as that is
            // handled in ApplyV3Rules for single transaction packages.
            // Future package RBF would conflict with the parent as well,
            // making this check sufficient.
            
```

Reply:
My view is that we'll end up reworking things substantially in order to enable package RBF, but I agree with the code comment suggestion.

Reply:
@sdaftuar do you mean for https://github.com/bitcoin/bitcoin/pull/28984 or for generalized package RBF? I'm thinking shorter term here.

Reply:
I don't think this is ever relevant in v3, even with future changes.

If your mempool parent has another mempool child, it doesn't matter if you have {RBF, sibling eviction, package RBF} within a package. Enumerating the possibilities in 1p1c:
```suggestion
            // The mempool or in-package parent cannot have any other in-mempool children. Since this is
            // a package, we don't need to check whether this can be a to-be-replaced sibling.
            // In a package, there must be a v3 violation even if a replacement is happening. Either:
            // - Package parent is replacing something and has both mempool parent and package child.
            // - Package child is replacing something and has both mempool parent and package parent.
```

Reply:
not sure I like this suggested text. anyone else want to give it a shot?

Reply:
I've made it "We don't check whether the sibling is to-be-replaced (done in ApplyV3Rules) because that doesn't apply in a package."